server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -234,9 +234,10 @@ private void runTasks(TaskInputs taskInputs) {
             }
             final long publicationStartTime = threadPool.relativeTimeInMillis();
             try {
-                ClusterChangedEvent clusterChangedEvent = new ClusterChangedEvent(summary, newClusterState, previousClusterState);
+                ClusterStatePublicationEvent clusterStatePublicationEvent =
+                    new ClusterStatePublicationEvent(summary, previousClusterState, newClusterState);
                 // new cluster state, notify all listeners
-                final DiscoveryNodes.Delta nodesDelta = clusterChangedEvent.nodesDelta();
```

Comment:
No point in holding the nodes delta as a field, we only use it here, so may as well throw it away once we're done with it.